scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb

```diff
@@ -104,22 +94,45 @@ namespace <%= namespace %>.SDK {
 
             return attributes;
         }
-
-        private void OnAttributesChange() {
-            _IsSaved = false;
-
-            OnChange(this);
-        }
     }
 
     /// <summary>
     /// Is used to add, update, or delete line items in a <see ref="Cart">Cart </see>.
     /// </summary>
     public class CartLineItems {
+        public static List<CheckoutLineItemInput> ConvertToCheckoutLineItemInput(List<CartLineItem> lineItems) {
+            var converter = new Converter<CartLineItem, CheckoutLineItemInput>((lineItem) => {
+                return lineItem.GetCheckoutLineItemInput();
+            });
+
+            return lineItems.ConvertAll(converter);
+        }
+
+        public static List<string> ConvertToLineItemIds(List<CartLineItem> lineItems) {
+            var converter = new Converter<CartLineItem, string>((lineItem) => {
+                return lineItem.ID;
+            });
+
+            List<string> lineItemIds = lineItems.ConvertAll(converter);
+
+            lineItemIds.RemoveAll(id => id == null);
+
+            return lineItemIds;
+        }
+
+        public bool IsSaved {
+            get {
+                return _IsSaved;
+            }
+        }
+
+        private bool _IsSaved = false;
         private List<CartLineItem> LineItems = new List<CartLineItem>();
-        private List<CartLineItem> LineItemsNew = new List<CartLineItem>();
-        private List<CartLineItem> LineItemsModified = new List<CartLineItem>();
-        private List<CartLineItem> LineItemsDeleted = new List<CartLineItem>();
+        private Action<string> OnDeleteLineItem;
```

Comment:
Some of these seem like they might benefit from events over single instance callbacks. With events, we aren't limiting the api to a single event consumer, and we could end up making it easier to interface with especially for view-style components that only need to know when to refresh their information.